sync/v1/sync.proto

```diff
@@ -1,25 +1,107 @@
 syntax = "proto3";
 
 import "google/protobuf/empty.proto";
-import "google/protobuf/any.proto";
 
 package protocol.sync.v1;
 
 option go_package = "github.com/harmony-development/legato/gen/sync/v1";
 
-message SyncRequest {
-  string token = 1;
-  string host = 2;
-}
+message Event {
+  message UserRemovedFromGuild {
+    uint64 user_id = 1;
+    uint64 guild_id = 2;
+  }
+  message UserAddedToGuild {
+  	uint64 user_id = 1;
+  	uint64 guild_id = 2;
+  }
 
-message PostBoxEvent { google.protobuf.Any event = 1; }
+  oneof kind {
+    UserRemovedFromGuild user_removed_from_guild = 1;
+    UserAddedToGuild user_added_to_guild = 2;
+  }
+}
 
 message PostEventRequest {
-  SyncRequest sync_request = 1;
-  PostBoxEvent event = 2;
+  Event event = 1;
+}
+
+// Acknowledgement of an event pulled using Pull.
+message Ack {
+  uint64 event_id = 1;
 }
 
+// A synchronisation message pulled using Pull.
+message Syn {
+  uint64 event_id = 1;
+  Event event = 2;
+}
+
+// # Postbox
+//
+// The postbox service forms the core of Harmony's server <-> server communications.
+//
+// It concerns the transfer of Events between servers, as well as ensuring reliable
+// delivery of them.
+//
+// The semantics of events are documented in the event types. The postbox service
+// is solely reliable for reliable pushing and pulling.
+//
+// ## Authorisation
+//
+// Requests are authorised using a JWT token in the Authorization HTTP header.
+//
+// The JWT token is signed using SHA-RSA-256 with the homeserver's private key,
+//
+// It contains the following fields, described using Go JSON semantics:
+// ```
+// Self string
+// Time uint53
+// ```
+//
+// Self is the server name of the server initiating the transaction. For Pull,
+// this tells the server being connected to which homeservers' events it should send.
+// For Push, this tells the server being connected to which homeservers' events it is
+// receiving.
+//
+// Time is the UNIX time in seconds of when the request is started. Servers should reject
```

Comment:
should probably clarify that this should not be relative to your time, rather UTC